### PR TITLE
s3: add DisableMultipart option

### PR DESCRIPTION
**Pull request by fatpat**
* [X] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* s3: add DisableMultipart option to force disabling multipart all together.

## Verification
Checked that the uploaded object was not upload with multipart

### CODE REVIEW

**Comment by fatpat** [comment]
anyone to look at this please ?

**Comment by fatpat** [comment]
I'd like a review if that's possible please :) 
thx

@yeya24 @saswatamcode @brancz

**Comment by yeya24** [comment]
Hi @fatpat, thanks for the contribution. May I know the usecase for disabling it? And maybe it is worth checking the conflicts since I just merged another pr.

**Comment by fatpat** [comment]
> Hi @fatpat, thanks for the contribution. May I know the usecase for disabling it? And maybe it is worth checking the conflicts since I just merged another pr.

Hello,

I just rebase.

The motivation comes from a need in mimir that use objstore as a dependency. For the motivation I'll copy the justifcation I wrote on the [mimir PR 7350](https://github.com/grafana/mimir/pull/7350):

> we discovered that we could achieve better GET performances from our internal objecstorage with multipart disabled.
> 
> our use case is a high load with a high frequency (1s interval). We were trying to play with the part size while we discovered it was not possible do disable it all together (thanos-io/objstore [resets part size if the object size is higher than the part size](https://github.com/thanos-io/objstore/blob/main/providers/s3/s3.go#L485C16-L486C1)).
> 
> Moreover all S3 clients have an option to disable multi-part uploads. Not having the option in mimir looks like something was missing.

**Review by yeya24 — APPROVED** [review]
Thanks

**Comment by fatpat** [comment]
> Thanks

Thanks to you for approving.

Who can click the merge button? 👌

**Comment by yeya24** [comment]
I am waiting to see if I can get another pair of eyes on this...
Maybe @fpetkovski @kakkoyun @saswatamcode?

File: README.md
Comment (fpetkovski) [inline README.md:184]: A bit unfortunate that we have to use a double negation, but I don't see another way to preserve backwards compatibility.

**Review by fpetkovski — APPROVED** [review]

**Comment by fatpat** [comment]
any news on this ? @yeya24  ?

**Review by yeya24 — APPROVED** [review]

**Comment by yeya24** [comment]
Sorry I forgot to merge it. Thanks for the reminder.

**Comment by fatpat** [comment]
> Sorry I forgot to merge it. Thanks for the reminder.

do you want me to rebase the PR ?

**Comment by yeya24** [comment]
I am not sure why this pr didn't get auto merged. Maybe rebase will help

**Comment by fatpat** [comment]
> I am not sure why this pr didn't get auto merged. Maybe rebase will help

done

**Pull request merged**
